pages/activities/[id]/index.js

```diff
@@ -14,11 +18,12 @@ export default function ActivityDetailsPage({ activities, onToggleBookmark }) {
 
```

Comment:
Ensuring that the modal's new state is set before navigating away can help prevent bugs.
 
 So 
```
  setShowModal(false);
  router.push("/");
```
 
 instead of 
```
  router.push("/");
  setShowModal(false);
```